conda_forge_tick/executors.py

```diff
@@ -21,16 +21,30 @@ def __exit__(self, *args, **kwargs):
 DRLOCK = DummyLock()
 
 
+@contextlib.contextmanager
+def lock_git_operation():
+    """
+    A context manager to lock git operations - it can be acquired once per thread, once per process,
+    and once per dask worker.
+    Note that this is a reentrant lock, so it can be acquired multiple times by the same thread/process/worker.
+    """
+
+    with TRLOCK, PRLOCK, DRLOCK:
+        yield
+
+
 logger = logging.getLogger(__name__)
 
 
 class DaskRLock(DaskLock):
     """A reentrant lock for dask that is always blocking and never times out."""
 
-    def acquire(self):
-        if not hasattr(self, "_rcount"):
-            self._rcount = 0
+    def __init__(self, name: str):
+        super().__init__(name=name)
+        self._rcount = 0
+        self._rdata = None
```

Comment:
This init signature is wrong. Please see the dask documentation. I specifically avoided overriding the init so I didn't have to keep the init here up to date with changes in dask. Please roll back the changes to this class and use the `getattr` and `setattr` approach.

Reply:
The only place where `DaskRLock` is instantiated is our own code (in `_init_dask`) - so there should be no problem with changing the `__init__` signature. There are readability benefits when all attributes are defined in `__init__`.

Reply:
Please back these changes out. We've initiated the dask lock with a client in the past.

Reply:
How about this (see my latest commit)?

Reply:
Yes that will work. Thank you!